reference/shaders-hlsl/comp/ssbo-array.comp

```diff
@@ -0,0 +1,25 @@
+struct SSBO
+{
+    float4 data[1];
+};
+
+RWStructuredBuffer<SSBO> ssbos : register(u0);
+
+static uint3 gl_GlobalInvocationID;
+struct SPIRV_Cross_Input
+{
+    uint3 gl_GlobalInvocationID : SV_DispatchThreadID;
+};
+
+void comp_main()
+{
+    uint ident = gl_GlobalInvocationID.x;
+    ssbos[1].data[ident] = ssbos[0].data[ident];
```

Comment:
So ... GLSL, and SPIR-V support scenarios like these:

```
buffer SSBO
{
    vec4 a;
    vec4 b;
    vec2 other_data[]; // Unsized array, can only be at the end.
};
```

The way I see it now, we can support the case where we do not have any unsized arrays by using the [0] trick.

A lone unsized array can only be supported directly because this is basically what structured buffers are ...

The combination however is a bit murky. I was under the impression that we could "pretend" that other_data is just an array of size 1, and we just access the array beyond that, but apparently that won't work?